src/fireedge/src/modules/components/Tables/VrTemplates/table.js

```diff
@@ -0,0 +1,303 @@
+        !fetchingMarketplaceApps &&
+        !fetchingDatastores
+      ),
+    [fetchingMarketplaces, fetchingMarketplaceApps, fetchingDatastores]
+  )
+
+  const handleDownloadDefaultImage = debounce(
+    async () => {
+      // Use cache if exists
+      setDisableDownload(true)
+      await fetchMarketplaces(undefined, true)
+      await fetchDatastores(undefined, true)
+      if (defaultImage !== null) {
+        setDialogOpen(true)
+      }
+    },
+    4000,
+    { leading: true }
+  )
+
+  useEffect(() => {
+    if (marketplaces?.length) {
+      const mid =
+        marketplaces?.find(
+          (marketplace) => marketplace?.NAME === 'OpenNebula Public'
+        ) ?? null
+      mid !== null &&
+        setOneMarket({
+          ID: mid?.ID ?? -1,
+          STATE: mid?.STATE === '0' ? 'ENABLED' : 'DISABLED',
+        })
+    }
+  }, [marketplaces])
+
+  useEffect(() => {
+    // Internal ASYNC funcs -->
+
+    const fetchApps = async () => {
+      await fetchMarketplaceApps(undefined, true) // Prefer cache
+    }
+
+    const enableMarket = async (marketID) => {
+      const response = await enableMarketplace({ id: marketID })
+      if (response?.data === parseInt(marketID, 10)) {
+        enqueueInfo(T.InfoEnableOpenNebulaMarketplace)
+        setOneMarket({ ...oneMarket, STATE: 'ENABLED' })
+        fetchApps()
+      }
+
+      return response
+    }
+
+    // <--Internal ASYNC funcs
+
+    try {
+      if (
+        oneMarket !== null &&
+        oneMarket?.ID !== -1 &&
+        oneMarket?.STATE !== 'ENABLED'
+      ) {
+        enableMarket(oneMarket?.ID).catch(console.error)
+
+        // Prefer cache
+      } else if (
+        oneMarket !== null &&
+        oneMarket?.ID !== -1 &&
+        oneMarket?.STATE === 'ENABLED'
+      ) {
+        fetchApps()
+      }
+    } catch {}
+  }, [oneMarket])
+
+  useEffect(() => {
+    if (marketplaceApps?.length && !fetchingMarketplaceApps) {
+      const vrTemplate =
+        marketplaceApps?.find(
+          (app) => app?.NAME === 'Service Virtual Router'
+        ) ?? null
+      setDefaultImage(vrTemplate)
+    }
+  }, [marketplaceApps])
+
+  useEffect(() => {
+    if (defaultImage !== null && !fetchingDatastores) {
+      setDisableDownload(true)
+      setDialogOpen(true)
+    }
+  }, [defaultImage, datastores])
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false)
+    setDisableDownload(false)
+  }
+
+  const handleDownloadApp = async (datastoreId) => {
+    await exportApp({
+      id: defaultImage?.ID,
+      name: 'Service Virtual Router',
+      vmname: 'Service Virtual Router',
+      datastore: datastoreId,
+    })
+      .then((res) => {
+        if (res?.data) {
+          enqueueSuccess(T.SuccessDownloadDefaultImage) && refetch()
+        } else if (res?.error) {
+          enqueueError(res.error)
+        } else {
+          throw new Error('Unexpected problem occured')
+        }
+      })
+      .catch((error) => {
+        console.error(error)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Log Injection occurs when untrusted user input is directly written to log files without proper sanitization. This can allow attackers to manipulate log entries, potentially leading to security issues like log forging or cross-site scripting. To prevent this, always sanitize user input using `encodeURIComponent()` or `DOMPurify.sanitize()` before logging.  Learn more - https://cwe.mitre.org/data/definitions/117.html